js/src/momoa.d.ts

```diff
@@ -0,0 +1,113 @@
+
+//-----------------------------------------------------------------------------
+// Options
+//-----------------------------------------------------------------------------
+
+type MomoaMode = "json" | "jsonc";
+
+interface MomoaTokenizeOptions {
```

Comment:
This is more of a personal preference, but I generally suggest to use `type` instead of `interface`:

```ts
type MomoaTokenizeOptions = {
	mode: MomoaMode;
	ranges: boolean;
};
```

And for interfaces that `extend` other type, you do it like this:

```ts
export type MomoaDocumentNode = MomoaNode & {
	body: MomoaNode;
};
```

Reply:
That's interesting. Can you explain more about why you prefer that?

Reply:
A few reasons:
1. Flexibility: If I want to rework the type later to an intersection of more types instead of having the props on it, the change is pretty simple without having an awkward empty interface that extends multiple types.
2. Simplicity: Using only type for type declarations makes the code simpler (only one reserved word for that instead of two). On top of that, the interface syntax is kinda unique, while the type syntax is used everywhere.
3. Shorter keyword: This is the least important for me, but is a nice plus. Simply "type" is less than half the keystrokes that "interface" has. Same for "extends" vs "&".